### PR TITLE
training script for knowledge distillation

### PR DESCRIPTION
**Pull request by paudelnirajan**
this is the refactored version of the training script for distillation that i experimented with: [this notebook](https://colab.research.google.com/drive/12arQzE0wu71O305bcPsug8z066LcVTnH?usp=sharing).

### CODE REVIEW

File: scripts/distill_nepali.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        quantization_config=bnb_config,
+        device_map="auto"
+    )
+    teacher.eval()
+    
+    # Load student (from checkpoint or fresh)
+    if args.resume_from and os.path.exists(args.resume_from):
+        print(f"Resuming from checkpoint: {args.resume_from}")
+        student = AutoModelForCausalLM.from_pretrained(
+            args.resume_from, torch_dtype=dtype, device_map="auto"
```
[the remark this answers is not in the document]
Comment (michaelpginn) [reply scripts/distill_nepali.py]: i.e. just have a single `student_model` arg that can be either a HF path or a checkpoint path

File: scripts/distill_nepali.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    
+    # Training loop
+    step = start_step
+    target_step = start_step + args.max_steps
+    accum_loss = 0.0
+    start_time = time.time()
+    
+    print(f"Training from step {start_step} to {target_step}")
+    
+    while step < target_step:
```
Comment (michaelpginn) [inline scripts/distill_nepali.py]: Right now the training loop can only go for one epoch, and doesn't shuffle. I'm wondering if we would need these. In that case, the dataset would have to be not streaming (for shuffling purposes) and you'll probably want to refactor a bit so that the batching/tokenization is done *prior* to the training loop. 

My usual approach is to do `dataset.map()` with your tokenizer ahead of time, and then use a Torch `DataLoader` to handle batching/shuffling.
